Hex updater crashes with FunctionClauseError in mix_hex_registry.key/1 when using private hex repository

### Is there an existing issue for this?

- [x] I have searched the existing issues

Related: #12539, #13940

### Package ecosystem

hex

### Package manager version

hex 2.3.1 (Dependabot's internal version, as shown in the stack trace)

### Language version

Elixir 1.19.5 (Dependabot's internal version)

### Manifest location and content before the Dependabot update

`/backend/mix.exs` and `/backend/mix.lock` in a private repository.

### dependabot.yml content

```yaml
version: 2

registries:
  oban-private-repo:
    type: hex-repository
    repo: oban
    url: https://repo.oban.pro
    auth-key: ${{ secrets.OBAN_LICENSE_KEY }}
    public-key-fingerprint: "SHA256:4/OSKi0NRF91QVVXlGAhb/BIMLnK8NHcx/EWs+aIWPc"

updates:
  - package-ecosystem: "mix"
    directory: "/backend"
    insecure-external-code-execution: allow
    registries:
      - "oban-private-repo"
    schedule:
      interval: "daily"
    commit-message:
      prefix: "chore(deps)"
    groups:
      oban:
        patterns:
          - "oban*"
```

### Updated dependency

All hex dependencies fail, not specific to one package. The error occurs for both packages from the private Oban registry (e.g. `oban_pro`) and public hex packages (e.g. `hackney`).

### What you expected to see, versus what you actually saw

**Expected:** Dependabot checks for outdated hex dependencies and creates PRs as it did through Feb 4, 2026.

**Actual:** Since Feb 5, 2026, every hex updater run crashes with:

```
FunctionClauseError: no function clause matching in :mix_hex_registry.key/1
```

The function at `mix_hex_registry.erl:159` receives HTTP response headers (a map with keys like `"accept-ranges"`, `"cache-control"`, `"content-length"`, etc.) instead of the expected registry public key data structure. This then causes `:erlang.binary_to_term("")` to fail with an `ArgumentError`.

**Stack trace:**

```
(FunctionClauseError) no function clause matching in :mix_hex_registry.key/1
  (hex 2.3.1) src/mix_hex_registry.erl:159: :mix_hex_registry.key(
    %{"accept-ranges" => "bytes", "age" => "175494", "cache-control" => "public, max-age=31536000, immutable",
      "connection" => "close", "content-length" => "427", "date" => "Wed, 25 Feb 2026 16:37:08 GMT",
      "fly-request-id" => "02KJ5K70WXYE9PJK2Z12KY706-ord", "server" => "Fly/84caf4a9 (2026-02-18)",
      "strict-transport-security" => "max-age=300", "vary" => "accept-encoding",
      "via" => "1.1 fly.io, 1.1 fly.io, 1.1 varnish", "x-cache" => "HIT", "x-cache-hits" => "0", ...})
  (hex 2.3.1) src/mix_hex_registry.erl:149: :mix_hex_registry.verify/3
  (hex 2.3.1) src/mix_hex_registry.erl:134: :mix_hex_registry.decode_and_verify_signed/2
  (hex 2.3.1) src/mix_hex_repo.erl:202: :mix_hex_repo.decode/4
  (hex 2.3.1) src/mix_hex_repo.erl:186: :mix_hex_repo.get_protobuf/3
  (hex 2.3.1) lib/hex/registry/server.ex:331: anonymous fn/3 in Hex.Registry.Server.prefetch_online/2

** (ArgumentError) errors were found at the given arguments:
  * 1st argument: invalid external representation of a term
    :erlang.binary_to_term("")
```

This fails 100% of the time. The updater retries repeatedly for ~25 minutes and crashes on every attempt with the same error.

### Native package manager behavior

`mix deps.update --all` works locally with the same Oban credentials and registry configuration. No issues fetching from `repo.oban.pro`.

### Images of the diff or a link to the PR, issue, or logs

Private repo so can't link directly, but the Dependabot job log shows the error repeating 10+ times before the run fails with:

```
Failure running container: Error: Command failed with exit code 1: /bin/sh -c $DEPENDABOT_HOME/dependabot-updater/bin/run update_files
##[error]Dependabot encountered an error performing the update
```

### Smallest manifest that reproduces the issue

Any `mix.exs` project that uses the Oban private hex repository (`repo.oban.pro`) with `oban_pro` as a dependency should reproduce this. The key part of the config is having a `hex-repository` registry pointing to `https://repo.oban.pro` with `public-key-fingerprint` and `auth-key`.

--- a/hex/helpers/lib/run.exs
+++ b/hex/helpers/lib/run.exs
@@ -6,6 +6,7 @@ defmodule DependencyHelper do
     |> case do
       {output, 0} ->
         output = try_decode(output)
+
         if output =~ "No authenticated organization found" do
           {:error, output}
         else
@@ -121,14 +122,14 @@ defmodule DependencyHelper do
 
   defp fetch_public_key(repo, repo_url, auth_key, fingerprint) do
     case Hex.Repo.get_public_key(%{trusted: true, url: repo_url, auth_key: auth_key}) do
-      {:ok, {200, key, _}} ->
+      {:ok, {200, _headers, key}} ->
         if public_key_matches?(key, fingerprint) do
           {:ok, key}
         else
           {:error, "Public key fingerprint mismatch for repo \"#{repo}\""}
         end
 
-      {:ok, {code, _, _}} ->
+      {:ok, {code, _headers, _body}} ->
         {:error, "Downloading public key for repo \"#{repo}\" failed with code: #{inspect(code)}"}
 
       other ->

